app/pages/lab/actions/media.js

```diff
@@ -5,17 +5,27 @@ const putFile = require('../../../lib/put-file');
 const MAX_FILE_SIZE = 500 * 1024;
 
 const mediaActions = {
-  fetchMedia(props = this.props) {
+  fetchMedia(props = this.props, page = 1) {
     this.setState({ media: null });
-
-    return props.resource.get(this.props.link, { page_size: this.props.pageSize })
+    
+    const page_size = props.pageSize;
+    
+    return props.resource.get(props.link, { page, page_size })
       .then((media) => {
-        return media.filter((medium) => {
-          return Object.keys(medium.metadata).length > 0;
-        });
+        const meta = media.length ? media[0].getMeta() : {};  // Derive the paging metadata for all the media, from the first media item.
+        return {
+          meta,
+          filteredMedia: media.filter((medium) => {
+            return Object.keys(medium.metadata).length > 0;
+          }),
+        };
       })
-      .then((filteredMedia) => {
-        this.setState({ media: filteredMedia });
+      .then((data) => {
+        this.setState({
+          page,
+          page_count: data.meta.page_count || 1,
+          media: data.filteredMedia,
+        });
       }).catch((error) => { return []; });
```

Comment:
I've refactored the code so that the `.catch()` block is in the right place. 👌 

That said, looking into the Media actions further, I realise that the `fetchMedia()` action doesn't need to return anything, _and doesn't perform any error handling,_ so the second then() and the catch() might be superfluous. 🤷 

I think I can rewrite the whole `fetchAction()` from `get().then().catch().then()` into a simpler `get().then()` chain, actually. Everything will be functionally the same, but I'm resisting the temptation to "improve" the lab code more than necessary. 😬 There lies a rabbit hole!